### PR TITLE
Microsoft authentication doc clarifications

### PR DESCRIPTION
**Pull request by afscrome**
* Add API Permissions to the app registration setup
* Encouraged re-use of an existing app registration over creating a new one.
* Moved the `Adding the provider to the Backstage front end` higher up so it's not so easily missed

Updated the Microsoft authentication document to include details on adding API Permissions. At more relaxed companies, these permission will be silently added to this list the first time they're requested, however in more locked down companies, a directory admin will need to consent to these permissions before users can use them.

Also slightly reworked the docs to encourage re-using any existing app registration.  This includes renaming the `AUTH_MICROSOFT_*` variables to match the `AZURE_*` ones used throughout other azure integrations.

This PR is not directly related the 1.19.1 patch, although debugging that did involve digging into the various consent options, which in turn found these holes in the docs.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/auth/microsoft/provider.md [hunk 1]
```diff
@@ -8,21 +8,36 @@ description: Adding Microsoft Azure as an authentication provider in Backstage
...
+- `offline_access`
+- `openid`
+- `profile`
+- `User.Read`
+
+Your company may require you to grant [admin consent](https://learn.microsoft.com/en-us/azure/active-directory/manage-apps/user-admin-consent-overview) for these permissions.
+Even if your company doesn't require admin consent, you may wish to do so as it means users don't need to individually consent the first time they access backstage.
+To grant admin consent, a directory admin will need to come to this page and click on the **Grant admin consent for COMPANY NAME** button.
+
+![App Registration Permissions](permissions.png)
```
Comment (awanlin) [inline docs/auth/microsoft/provider.md:36]:
```suggestion
![App Registration Permissions](./permissions.png)
```
When I use the rich preview this is a broken image for me, wondering if you need the `./` as suggested above

File: docs/auth/microsoft/provider.md [hunk 2]
```diff
@@ -8,21 +8,36 @@ description: Adding Microsoft Azure as an authentication provider in Backstage
...
+## Configure App Registration on Azure
+
+Depending on how locked down your company is, you may need a directory administrator to do some or all of these instructions.
+
+Go to [Azure Portal > App registrations](https://portal.azure.com/#view/Microsoft_AAD_RegisteredApps/ApplicationsListBlade) and find your existing app registration, or create a new one.
+If you have an existing App Registration for Backstage, use that rather than create a new one.
+
+On your app registration's overview page, add a new `Web` platform configuration, with the configuration:
+
+- **Redirect URI**: `http://your-backstage.com/api/auth/microsoft/handler/frame` (for local dev, typically `http://localhost:7007/api/auth/microsoft/handler/frame`)
```
Comment (awanlin) [inline docs/auth/microsoft/provider.md:20]:
```suggestion
- **Redirect URI**: `https://your-backstage.com/api/auth/microsoft/handler/frame` (for local dev, typically `http://localhost:7007/api/auth/microsoft/handler/frame`)
```

**Review by awanlin — CHANGES_REQUESTED** [review]
Looks good @afscrome, left a pair of minor comments.

While you have this open would it make sense to start adding mention to Microsoft Entra ID? Totally optional, will leave that for you to decide

File: docs/auth/microsoft/provider.md (hunk 1 above)
Comment (awanlin) [reply docs/auth/microsoft/provider.md:36]: Never mind, looks fine when I view it in your branch 👍

**Comment by afscrome** [comment]
Re the Entra ID rename, I did start but it blew up in scope so I think it makes sense for a separate PR.

Microsoft have a handy script to help with the rename, although it will also need some manual work on top as we'll want to leave a few `formerly Azure Active Directory` phrases around as not everyone is familiar with the new name, and we'll also want the docs to show up if someone searches for aad.
 https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/how-to-rename-azure-ad#sample-powershell-script

**Review by awanlin — APPROVED** [review]
Awesome, thanks again @afscrome for taking the time to make these changes, they have a huge positive impact! 🚀 

For the name change thanks for the details and for taking a try at it, I agree it's probably best for its own PR 👍

**Review by vinzscam — APPROVED** [review]

File: docs/auth/microsoft/provider.md [hunk 3]
```diff
@@ -35,9 +50,9 @@ auth:
   providers:
     microsoft:
       development:
-        clientId: ${AUTH_MICROSOFT_CLIENT_ID}
-        clientSecret: ${AUTH_MICROSOFT_CLIENT_SECRET}
-        tenantId: ${AUTH_MICROSOFT_TENANT_ID}
+        clientId: ${AZURE_CLIENT_ID}
+        clientSecret: ${AZURE_CLIENT_ID}
```
Comment (grvpandey11) [inline docs/auth/microsoft/provider.md:54]: Isn't it should be AZURE_CLIENT_SECRET

**Review by grvpandey11 — CHANGES_REQUESTED** [review]

File: docs/auth/microsoft/provider.md (hunk 3 above)
Comment (awanlin) [reply docs/auth/microsoft/provider.md:54]: Good catch!

Comment (afscrome) [reply docs/auth/microsoft/provider.md]: 🤦 absolutely

**Review by benjdlambert — APPROVED** [review]
Nice! thanks :tada"

**Pull request merged**